common/test/acceptance/pages/studio/html_component_editor.py

```diff
@@ -7,25 +7,59 @@ class HtmlComponentEditorView(ComponentEditorView):
     """
     Represents the rendered view of an HTML component editor.
     """
-    def set_content_and_save(self, content):
-        """
-        Types content into the html component and presses Save.
+    def set_content_and_save(self, content, raw=False):
+        """Types content into the html component and presses Save.
+
+        Arguments:
+            content (str): The content to be used.
+            raw (bool): If true, edits in 'raw HTML' mode.
         """
-        self.set_content(content)
+        if raw:
+            self.set_raw_content(content)
+        else:
+            self.set_content(content)
+
         self.save()
 
-    def set_content_and_cancel(self, content):
-        """
-        Types content into the html component and presses Cancel to abort the change.
+    def set_content_and_cancel(self, content, raw=False):
+        """Types content into the html component and presses Cancel to abort.
+
+        Arguments:
+            content (str): The content to be used.
+            raw (bool): If true, edits in 'raw HTML' mode.
         """
-        self.set_content(content)
+        if raw:
+            self.set_raw_content(content)
+        else:
+            self.set_content(content)
+
         self.cancel()
 
     def set_content(self, content):
-        """
-        Types content into the html component, leaving the component open.
+        """Types content into the html component, leaving the component open.
+
+        Arguments:
+            content (str): The content to be used.
         """
         self.q(css='.edit-xblock-modal .editor-modes .editor-button').click()
-        editor = self.q(css=self._bounded_selector('.html-editor .mce-edit-area'))[0]
+
+        selector = '.html-editor .mce-edit-area'
+        editor = self.q(css=self._bounded_selector(selector))[0]
         ActionChains(self.browser).click(editor).\
-            send_keys([Keys.CONTROL, 'a']).key_up(Keys.CONTROL).send_keys(content).perform()
+            send_keys([Keys.CONTROL, 'a']).key_up(Keys.CONTROL).\
+            send_keys(content).perform()
+
+    def set_raw_content(self, content):
+        """Types content in raw html mode, leaving the component open.
+
+        Arguments:
+            content (str): The content to be used.
+        """
+        self.q(css='.edit-xblock-modal .editor-modes .editor-button').click()
```

Comment:
Nit: can you define a variable for this CSS (repeated at line 44).

I'm also tempted to have a helper method for doing the select-all.

You could solve the situation unifying the 2 set_content methods, with a raw boolean. Seems like the only differences are lines 59 and 65.